server/src/thread/model.rs

```diff
@@ -350,6 +351,95 @@ impl Threads {
 
         Ok(comment)
     }
+
+    pub async fn update_comment<'a, E>(
+        payload: &update_comment_mutation::Payload,
+        executor: E,
+    ) -> Result<operation::SuccessOutput, operation::Error>
+    where
+        E: Executor<'a>,
+    {
+        if payload.content.is_empty() {
+            return Err(operation::Error::BadRequest {
+                reason: "content is empty".to_string(),
+            });
+        }
+
+        let mut transaction = executor.begin().await?;
+
+        let is_trashed = match sqlx::query!(
```

Comment:
Instead of using a `match` it is common to use `map_err` in this repo. 
I personally think it is easier to read with map_err